src/ray/gcs/tables.h

```diff
@@ -175,9 +175,23 @@ class Log : virtual public PubsubInterface<ID> {
   Status CancelNotifications(const JobID &job_id, const ID &id,
                              const ClientID &client_id);
 
+  void AddShards(std::vector<std::shared_ptr<RedisContext>> &contexts) {
+    if (shard_contexts_.size() > 0) {
+      return;  // Only add shards if at the first time.
```

Comment:
I'm not sure if we want to keep this method, it seems to be a sign that we're initializing things in the wrong order (which we probably are), but anyway, this should be `RAY_CHECK(shard_contexts_.size() == 0)` instead of the if statement I believe.